R/plots-qc.R

```diff
@@ -64,6 +64,78 @@ qc_sample_hist <- function(gimap_dataset, wide_ar = 0.75) {
   return(sample_cpm_histogram)
 }
 
+#' Create a histogram for the variance within replicates for each pgRNA
+#' @description This function uses pivot_longer to rearrange the data for plotting, finds the variance for each pgRNA construct (using row number as a proxy) and then plots a histogram of these variances
+#' @param gimap_dataset The special gimap_dataset from the `setup_data` function which contains the transformed data
+#' @param wide_ar aspect ratio, default is 0.75
+#' @importFrom tidyr pivot_longer
+#' @importFrom magrittr %>%
+#' @import ggplot2
+#' @return a ggplot histogram
+#' @examples \dontrun{
+#'
+#' }
+#'
+
+qc_variance_hist <- function(gimap_dataset, wide_ar = 0.75){
+  
+  return(
+    gimap_dataset$transformed_data$log2_cpm[,3:5] %>%
+      as.data.frame() %>%
+      mutate(row = row_number()) %>%
+      tidyr::pivot_longer(-row) %>%
+      group_by(row) %>%
+      summarize(var = var(value)) %>%
+      ggplot(aes(x=var)) +
+      geom_histogram(binwidth = 0.1) +
+      theme(panel.background = element_blank(),
+            panel.grid = element_blank(),
+            aspect.ratio = wide_ar) +
+      xlab("variance") +
+      ylab("pgRNA construct count")
+  )  
+  
+}
+
+#' Create a bar graph that shows the number of replicates with a zero count for pgRNA constructs flagged by the zero count filter
+#' @description A short description...
+#' @param gimap_dataset
+#' @param wide_ar aspect ratio, default is 0.75
+#' @importFrom tidyr pivot_longer
+#' @importFrom magrittr %>%
+#' @import ggplot2 
+#' @return a ggplot barplot
+#' @examples \dontrun{
+#' 
```

Comment:
Can you add examples here? Helps me review this if I know how you envision this function will be called.

Reply:
I found your function calls from the QC report and just brought them here. Correct these if this is not how this function could be called.